packages/panzer/disc-fe/src/Panzer_BasisValues2_impl.hpp

```diff
@@ -807,7 +807,12 @@ getBasisValuesRef(const bool cache,
   const int num_card  = basis_layout->cardinality();
 
   auto tmp_basis_ref_scalar = af.buildStaticArray<Scalar,BASIS,IP>("dyn_basis_ref_scalar",num_card,num_quad);
-  intrepid_basis->getValues(tmp_basis_ref_scalar.get_view(), cubature_points_uniform_ref_.get_view(), Intrepid2::OPERATOR_VALUE);
+
+  // FIXME: Intrepid does not appear to be const correct
+  auto nonconst_cub_points = af.buildStaticArray<Scalar,IP,Dim>("scratch: cub_points", cubature_points_uniform_ref_.extent(0), cubature_points_uniform_ref_.extent(1));
+  Kokkos::deep_copy(nonconst_cub_points.get_view(), cubature_points_uniform_ref_.get_view());
```

Comment:
Instead of allocating arrays and dong copies, why don't you pull out the data ptr and build an unmanaged view? Could do that for all of these calls to intrepid. @mperego - is this something we could look into next FY?

Reply:
Good call, I switched over, and will re-run on CUDA to make sure.